tests/aws/services/cloudformation/api/test_update_stack.py

```diff
@@ -407,3 +408,47 @@ def test_update_with_rollback_configuration(deploy_cfn_template, aws_client):
 
     # cleanup
     aws_client.cloudwatch.delete_alarms(AlarmNames=["HighResourceUsage"])
+
+
+@markers.aws.validated
+@markers.snapshot.skip_snapshot_verify(["$..Stacks..ChangeSetId"])
+def test_diff_after_update(deploy_cfn_template, aws_client, snapshot):
+    template_1 = textwrap.dedent("""
+    Resources:
+        SimpleParam:
+            Type: AWS::SSM::Parameter
+            Properties:
+                Value: before-stack-update
+                Type: String
+    """)
+    template_2 = textwrap.dedent("""
+    Resources:
+        SimpleParam1:
+            Type: AWS::SSM::Parameter
+            Properties:
+                Value: after-stack-update
+                Type: String
+    """)
+
+    stack = deploy_cfn_template(
+        template=template_1,
+    )
+
+    aws_client.cloudformation.get_waiter("stack_create_complete").wait(StackName=stack.stack_name)
+    aws_client.cloudformation.update_stack(
+        StackName=stack.stack_name,
+        TemplateBody=template_2,
+    )
+    aws_client.cloudformation.get_waiter("stack_update_complete").wait(StackName=stack.stack_name)
+    get_template_response = aws_client.cloudformation.get_template(StackName=stack.stack_name)
+    snapshot.match("get-template-response", get_template_response)
+
+    with pytest.raises(botocore.exceptions.ClientError) as exc_info:
+        aws_client.cloudformation.update_stack(
+            StackName=stack.stack_name,
+            TemplateBody=template_2,
+        )
+    snapshot.match("update-error", exc_info.value.response)
+
+    describe_stack_response = aws_client.cloudformation.describe_stacks(StackName=stack.stack_name)
+    assert describe_stack_response["Stacks"][0]["StackStatus"] == "UPDATE_COMPLETE"
```

Comment:
nit: this is a little confusing. The stack status is set to `UPDATE_COMPLETE` after the first stack update, not after the failed attempt to update the stack. Perhaps move this block to before the `pytest.raises` block?

Reply:
In LocalStack, we [manually set the stack status](https://github.com/localstack/localstack/pull/10556/files#diff-21252aafb0a3b77f8f2944c7405ef6321e37825908431516c86e19ffdddc51e7R379-R381) when no stack updates are necessary. So I wanted to verify that the status is indeed updated.

Reply:
Sure, but the stack was _already_ in `UPDATE_COMPLETE`